src/main/java/bg/codexio/springframework/boot/adapters/GraphQLComplexFilterAdapter.java

```diff
@@ -0,0 +1,44 @@
+package bg.codexio.springframework.boot.adapters;
+
+import bg.codexio.springframework.data.jpa.requery.payload.FilterGroupRequest;
+import bg.codexio.springframework.data.jpa.requery.payload.FilterRequestWrapper;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import jakarta.servlet.http.HttpServletRequest;
+
+/**
+ * An interface for adapting GraphQL-like complex filters from a JSON string
+ * to a {@link FilterRequestWrapper}.
+ * Implementations of this interface are responsible for handling and parsing
+ * GraphQL complex filters
+ * and providing them as {@link FilterRequestWrapper} instances.
+ */
+public interface GraphQLComplexFilterAdapter {
+    /**
+     * Determines whether this adapter supports the given
+     * {@link HttpServletRequest}.
+     * Currently, this method always returns {@code true}, but this behavior
+     * can be modified.
+     *
+     * @param req the HTTP servlet request to check
+     * @return {@code true} if the adapter supports the request, {@code false
+     * } otherwise
+     */
+    boolean supports(HttpServletRequest req);
```

Comment:
Will we ever use this for complex adaptation?

Reply:
Won't we when the add the dgraph/prisma adapters?